OpenSim/Simulation/SimbodyEngine/Test/testJoints.cpp

```diff
@@ -1946,11 +1946,12 @@ void testAutomaticJointReversal()
     //thigh.addGeometry(Cylinder(0.035, 0.4));
     thigh->upd_geometry(0).setColor(SimTK::Vec3(0, 0, 1));   // BLUE
     
-    //TODO fix Bug: ModleComponent::addGeometry assumes ownership but does not
-    // take a pointer to heap allocated Geometry
-    shank->addGeometry(*new Cylinder(0.02, 0.243800));
+    //ModelComponent::addGeometry makes a copy of the passed in Geometry
+    Cylinder cyl(0.02, 0.243800);
+    shank->addGeometry(cyl);
```

Comment:
Oh I'm so sorry! I was having trouble reading the diff. Thanks for fixing the leak.